File: scala-play-server/app/model/RequestTimeMapArrivalSearch.scala
package model

import play.api.libs.json._
import java.time.OffsetDateTime

/**
  * Represents the Swagger definition for RequestTimeMapArrivalSearch.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2020-07-14T11:29:17.777Z[Etc/UTC]")
case class RequestTimeMapArrivalSearch(
  id: String,
  coords: Coords,
  transportation: RequestTransportation,
  travelTime: Int,
  arrivalTime: OffsetDateTime,
  properties: Option[List[RequestTimeMapProperty]],
  range: Option[RequestRangeNoMaxResults]
)

object RequestTimeMapArrivalSearch {
  implicit lazy val requestTimeMapArrivalSearchJsonFormat: Format[RequestTimeMapArrivalSearch] = Json.format[RequestTimeMapArrivalSearch]
}

File: javascript-closure-angular/API/Client/RequestTimeMapArrivalSearch.js
goog.provide('API.Client.RequestTimeMapArrivalSearch');

/**
 * @record
 */
API.Client.RequestTimeMapArrivalSearch = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.id;

/**
 * @type {!API.Client.Coords}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.coords;

/**
 * @type {!API.Client.RequestTransportation}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.transportation;

/**
 * @type {!number}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.travelTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.arrivalTime;

/**
 * @type {!Array<!API.Client.RequestTimeMapProperty>}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.properties;

/**
 * @type {!API.Client.RequestRangeNoMaxResults}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.range;

/**
 * @type {!API.Client.RequestLevelOfDetail}
 * @export
 */
API.Client.RequestTimeMapArrivalSearch.prototype.levelOfDetail;

